File: app/src/main/java/era/apps/happinessjar/models/stories/view_model/StoriesViewModel.java
package era.apps.happinessjar.models.stories.view_model;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

import era.apps.happinessjar.models.stories.data_base.Story;

public class StoriesViewModel extends AndroidViewModel {
    private LiveData<List<Story>> allStories;

    public StoriesViewModel(@NonNull Application application) {
        super(application);
        StoryRepository storyRepository = new StoryRepository(application);
        allStories      = storyRepository.getAllStories();
    }

    public LiveData<List<Story>> getAllStories() {
        return allStories;
    }




}

File: app/src/main/java/era/apps/happinessjar/ui/viewmodel/StoriesViewModel.kt
package era.apps.happinessjar.ui.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import era.apps.happinessjar.data.models.Story
import era.apps.happinessjar.data.repository.StoryRepository

class StoriesViewModel(application: Application) : AndroidViewModel(application) {
    val allStories: LiveData<List<Story>>

    init {
        val storyRepository = StoryRepository(application)
        allStories = storyRepository.allStories
    }
}